cmd/agent/client/client.go

```diff
@@ -5,32 +5,34 @@ import (
 	"fmt"
 	"log"
 	"net/http"
+	"sync"
 
 	"github.com/go-resty/resty/v2"
 
+	"github.com/aykuli/observer/internal/agent/config"
 	"github.com/aykuli/observer/internal/models"
 	"github.com/aykuli/observer/internal/sign"
 
 	"github.com/aykuli/observer/internal/agent/storage"
 	"github.com/aykuli/observer/internal/compressor"
 )
 
-type MerticsClient struct {
+type MetricsClient struct {
 	ServerAddr string
 	memStorage *storage.MemStorage
 	signKey    string
 	limit      int
 }
 
-func NewMetricsClint(serverAddr string, memStorage *storage.MemStorage) *MerticsClient {
-	return &MerticsClient{
-		ServerAddr: serverAddr,
+func NewMetricsClient(config config.Config, memStorage *storage.MemStorage) *MetricsClient {
+	return &MetricsClient{
+		ServerAddr: "http://" + config.Address,
 		memStorage: memStorage,
+		signKey:    config.Key,
+		limit:      config.RateLimit,
 	}
 }
-
-// SendMetrics deprecated from iteration 5
-func (m *MerticsClient) SendMetrics(req *resty.Request) {
+func (m *MetricsClient) SendMetrics(req *resty.Request) {
```

Comment:
```suggestion
func (m *MetricsClient) SendMetrics(restyClient *resty.Client) {
```
Сразу не заметил, но у resty потокобезопасен только сам клиент. Если попытаться конкурентно воспользоваться объектом `resty.Request`, то попадаем на DATA RACE. Поэтому объект `resty.Request` надо генерить внутри `sendOneMetric` непосредственно перед отправкой, а потом уничтожать.

Reply:
нашла ишю, действительно